### ISSUE TITLE
AllowNamedArgumentOnlyInAttributesRule - issue with nested attributes

### ISSUE THREAD

**Issue by enumag** [shipmonk-rnd/phpstan-rules#151]
Consider this code:

```php
    #[OA\Response(
        response: '200',
        description: 'Lorem ipsum',
        content: new OA\JsonContent(
            ref: new Model(
                type: CustomResponseClass::class
            )
        )
    )]
```

Both the `ref:` and `type:` trigger "Named arguments are allowed only within native attributes" because here JsonContent and Model are not actually considered attributes as they're created using `new`. Personally though in this context I'd like to allow named arguments as they're nested attributes.

What do you think?

**Comment by enumag** [comment]
Ah nevermind. https://github.com/shipmonk-rnd/phpstan-rules/pull/136

**Comment by janedbal** [comment]
I would agee, but as linked, this one wont exist in v3. Feel free to copypaste and improve in your project if you see some value in it.
